### PR TITLE
Kogito 2532 - monitoring addon support multiple dmn built in types

### PR DESCRIPTION
**Pull request by r00ta**
Jira: https://issues.redhat.com/browse/KOGITO-2532

This PR adds the support of the remaining dmn built-in types in the prometheus-monitoring-addon. 
In particular, the remaining 5 time-based built in types are supported. For all of them, a graph with the quantiles is generated in the grafana dashboards (with different scale and unit measurement): 
- date and time: 
Screenshot from 2020-06-30 10-16-12
- days and time duration: 
Screenshot from 2020-06-30 10-16-17
- years and months duration: total number of months (`Period` - `toTotalMonths`)
Screenshot from 2020-06-30 10-16-30
- time: 
Screenshot from 2020-06-30 10-16-34
- date: 
Screenshot from 2020-06-30 10-16-41

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: grafana-api/src/main/java/org/kie/kogito/grafana/GrafanaConfigurationWriter.java [hunk 1]
```diff
@@ -95,10 +95,11 @@ public static String generateDomainSpecificDMNDashboard(String templatePath, Str
                     jgrafana.addPanel(PanelType.GRAPH,
                                       "Decision " + decision.getName(),
                                       String.format("%s_dmn_result{endpoint = \"%s\", decision = \"%s\"}",
-                                                    type,
+                                                    type.toString().replace(" ", "_"),
```
Comment (danielezonca) [inline grafana-api/src/main/java/org/kie/kogito/grafana/GrafanaConfigurationWriter.java:98]: What is the reason of this change? How this string is used?
Side note, are you sure `type` cannot be `null`?

Comment (r00ta) [reply grafana-api/src/main/java/org/kie/kogito/grafana/GrafanaConfigurationWriter.java:98]: @danielezonca dmn type might contain spaces (for example the type "time and date"), and in the PrompQL query of grafana spaces are not allowed. So by default we have to replace the spaces with something else

Comment (r00ta) [reply grafana-api/src/main/java/org/kie/kogito/grafana/GrafanaConfigurationWriter.java:98]: For your other question, the check on `null` is already done few lines above this one

**Comment by r00ta** [comment]
Hi @danielezonca @jiripetrlik , I think this PR is ready for the first iteration of reviews

File: addons/monitoring-prometheus-addon/src/test/java/org/kie/kogito/monitoring/integration/LocalDateTimeHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        handler = new LocalDateTimeHandler("hello", registry);
+    }
+
+    @AfterEach
+    public void destroy() {
+        registry.clear();
+    }
+
+    @Test
+    public void GivenLocalDateTimeMetrics_WhenMetricsAreStored_ThenTheQuantilesAreCorrect() {
```
[the remark this answers is not in the document]
Comment (r00ta) [reply addons/monitoring-prometheus-addon/src/test/java/org/kie/kogito/monitoring/integration/LocalDateTimeHandlerTest.java]: Sorry, old C# habits die hard :) Renamed

[the remark this answers is not in the document]
Comment (jiripetrlik) [reply addons/monitoring-prometheus-addon/src/test/java/org/kie/kogito/monitoring/integration/LocalDateTimeHandlerTest.java]: :-) Thank you!

File: addons/monitoring-prometheus-addon/src/main/java/org/kie/kogito/monitoring/system/metrics/dmnhandlers/TypeHandlerWithSummary.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.kogito.monitoring.system.metrics.dmnhandlers;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Summary;
+
+public abstract class TypeHandlerWithSummary<T> implements TypeHandler<T> {
```
[the remark this answers is not in the document]
Comment (r00ta) [reply addons/monitoring-prometheus-addon/src/main/java/org/kie/kogito/monitoring/system/metrics/dmnhandlers/TypeHandlerWithSummary.java]: Agreed, I've refactored it with an interface

**Comment by r00ta** [comment]
jenkins retest this please
